Add CTA to migrate to MB Cloud in admin panel

mazameli opened the pull request:
How this is looking so far:

![image](https://user-images.githubusercontent.com/2223916/105113424-c2c27a80-5a79-11eb-8dd3-53a703480568.png)

We talked about adding a CTA in the Email config page and the Admin checklist page as well.

Also, right now the button just links to `metabase.com/start/hosted`, but I thinking we're going to want a more specific landing page where we can talk about migrating specifically, like the mockups we had here, which shouldn't be hard to slap together at all:

![image](https://user-images.githubusercontent.com/2223916/105113550-03ba8f00-5a7a-11eb-9e94-f18de6100462.png)

mazameli committed:
add mb cloud cta to admin updates page

salsakran commented:
We should have a specific url that this links to that we could then redirect to an appropriate page from www, that way we won't have to worry about changing it or pushing new versions.

mazameli commented:
> We should have a specific url that this links to that we could then redirect to an appropriate page from www, that way we won't have to worry about changing it or pushing new versions.

Yep, makes a ton of sense. 👍🏼 I'll add that URL over on the website repo.

mazameli committed:
change url to the new redirect url

mazameli committed:
add cta to email setup form

mazameli committed:
add cta to admin checklist page

mazameli commented:
- Updated the redirect URL to match the PR on the website.
- Added CTAs to the email setup and admin checklist pages:

<img width="1457" alt="image" src="https://user-images.githubusercontent.com/2223916/105266216-04205c00-5b46-11eb-8487-da8c2d2284f1.png">


<img width="1454" alt="image" src="https://user-images.githubusercontent.com/2223916/105266050-f965c700-5b45-11eb-9c4b-addb8096698a.png">

mazameli committed:
add cta to state when update is available

paulrosenzweig reviewed (approved):
Looks good!

How are we going to suppress this for instances running in MB Cloud? Doesn't really matter but seems nice to do.

mazameli commented:
> How are we going to suppress this for instances running in MB Cloud? Doesn't really matter but seems nice to do.

Ah right, thanks for reminding me — I was going to ask around about how to do that. Any ideas? 'Cause yeah, we definitely want to hide these in hosted instances.

mazameli commented on frontend/src/metabase/admin/settings/components/SettingsSetupList.jsx:
```diff
@@ -101,28 +105,48 @@ export default class SettingsSetupList extends Component {
-              {tasks.map((section, index) => (
-                <TaskSection {...section} key={index} />
-              ))}
-            </div>
-          )}
-        </LoadingAndErrorWrapper>
-      </div>
+          <Icon name="cloud" size={48} style={{ color: "#B9D8F4" }} />
+          <div className="pb3">
+            <Text className="text-brand mb0">{t`Have your server maintained for you.`}</Text>
+            <Text className="text-brand text-bold">{t`Migrate to Metabase Cloud.`}</Text>
+          </div>
+
+          <ExternalLink
+            className="bordered rounded border-brand bg-brand-hover text-white-hover px2 py1 text-bold text-center"
+            href={"https://www.metabase.com/migrate/from/selfhosted"}
+            target="_blank"
+          >
+            {t`Learn more`}
+          </ExternalLink>
+        </div>
+      </Flex>
```
Yeah, I think it would make sense at least to do that with the ones that are in the side margin. The other one that's below the updates-check is too different stylistically.

paulrosenzweig replied:
It looks like they all share the same "Learn more" button so maybe one file that exports a MarginHostingCTA and a HostingInfoLink.

mazameli replied:
Good call. I went ahead and extracted the margin bit into `metabase/admin/settings/components/widgets/MarginHostingCTA.jsx` and the learn more button into `metabase/admin/settings/components/widgets/HostingInfoLink.jsx`

mazameli committed:
extract reusable bits into components

mazameli committed:
hide CTAs if site url is on mb cloud

mazameli commented:
Hiding the CTAs by looking at the site-url was surprisingly easy, but @paulrosenzweig, let me know if I should change anything about [how I implemented this](https://github.com/metabase/metabase/pull/14458/commits/e3f19204da2e16fd079d1f7fcf67d033f8586766). 

![hide-cta-in-action](https://user-images.githubusercontent.com/2223916/105536965-0c8fa880-5ca6-11eb-9f8c-49d3f761bfc1.gif)

dpsutton commented on frontend/src/metabase/admin/settings/components/SettingsEmailForm.jsx:
```diff
@@ -23,6 +24,10 @@ const SEND_TEST_BUTTON_STATES = {
   success: t`Sent!`,
 };
 
+const isHosted = MetabaseSettings.get("site-url")
```
might it be better to imitate the way some of the settings have an accessor?

```js
  adminEmail() {
    return this.get("admin-email");
  }
```

but for is hosted. then when the env var is exposed this can change in one place without the isHosted definiition duplicated in many places?

mazameli replied:
Ah, perfect — I was thinking along these lines but didn't know where to look. I'll amend.

mazameli committed:
extract isHosted logic to an accessor

mazameli commented:
I think I successfully extracted the isHosted logic to an accessor in the settings file now. The hide/show behavior is still working correctly. Where do I pick up my front-end engineer badge?

https://github.com/metabase/metabase/pull/14458/commits/f9cee75d977abbf5caaeb6e6757d3c17a32cac79

mazameli commented:
@salsakran Do me a favor and lmk how this looks to you. Should be able to merge it provided @paulrosenzweig doesn't see anything wrong with the code rejiggering I did.

paulrosenzweig reviewed (approved):
Code changes look good!

salsakran reviewed (approved):
looks solid to me, but I didn't pull this down and build it, just reviewing copy+locations.
